### PR TITLE
Catch http errors and pass on the error code, eg 404.

### PR DESCRIPTION
**Pull request by yemble**

### CODE REVIEW

File: packimg/packimg.rb [hunk 1]
```diff
@@ -3,9 +3,26 @@
...
+# HTTP requester with redirection following
+def http_fetch(uri_str, limit = 3)
+  raise ArgumentError, 'HTTP redirect too deep' if limit <= 0
+
+  uri = URI.parse(uri_str)
+  request = Net::HTTP::Get.new(uri.path, {'User-Agent' => 'MetaBroadcast image resizer'})
+  response = Net::HTTP.start(uri.host, uri.port) { |http| http.request(request) }
+
+  case response
+  when Net::HTTPRedirection then http_fetch(response['location'], limit - 1)
```
Comment (tsprlng) [inline packimg/packimg.rb]: <3 ruby

**Comment by tsprlng** [comment]
:+1: !!

File: packimg/packimg.rb (hunk 1 above)
Comment (yemble) [inline packimg/packimg.rb]: ಠ_ಠ

**Comment by tfm** [comment]
@yemble can we submit a pull request to upstream for this? seems like a fix that everyone would benefit from it, not just us!

**Pull request merged**